client/src/sagas/alert.js

```diff
@@ -0,0 +1,24 @@
+import { delay, put, takeEvery } from 'redux-saga/effects';
+import { uuid } from 'uuidv4';
+
+import { alert } from './types';
+
+export function* handleAddAlert({ payload }) {
```

Comment:
what is a `function*`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*

TL;DR - it's a generator function, which is a special function whose execution can be paused and resumed. Behind the scenes, I believe these functions are akin to objects, and they have a `next()` method to continue executing until it reaches a `yield`, upon which `next()` would need to be called again to continue execution.

Generators are what is used by our state management tool (sagas) to handle incoming actions, and they handle the details of this calling `next()` business.

Reply:
huh. neat. :til: thanks!